src/main/java/com/easypost/service/ClaimService.java

```diff
@@ -0,0 +1,107 @@
+package com.easypost.service;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.exception.General.EndOfPaginationError;
+import com.easypost.http.Requestor;
+import com.easypost.http.Requestor.RequestMethod;
+import com.easypost.model.Claim;
+import com.easypost.model.ClaimCollection;
+
+import lombok.SneakyThrows;
+
+public class ClaimService {
+    private final EasyPostClient client;
+
+    /**
+     * ClaimService constructor.
+     *
+     * @param client The client object.
+     */
+    ClaimService(EasyPostClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Create a new claim object from a map of parameters.
+     *
+     * @param params Map of parameters.
+     * @return Claim object
+     * @throws EasyPostException when the request fails.
+     */
+    public Claim create(final Map<String, Object> params) throws EasyPostException {
+        String endpoint = "claims";
+
+        return Requestor.request(RequestMethod.POST, endpoint, params, Claim.class, client);
+    }
+
+    /**
+     * Retrieve an Claim from the API.
+     *
+     * @param id The ID of the Claim to retrieve.
+     * @return Claim object
+     * @throws EasyPostException when the request fails.
+     */
+    public Claim retrieve(final String id) throws EasyPostException {
+        String endpoint = "claims/" + id;
+
+        return Requestor.request(RequestMethod.GET, endpoint, null, Claim.class, client);
+    }
+
+    /**
+     * Get a list of Claims.
+     *
+     * @param params a map of parameters
+     * @return ClaimCollection object
+     * @throws EasyPostException when the request fails.
+     */
+    public ClaimCollection all(final Map<String, Object> params) throws EasyPostException {
+        String endpoint = "claims";
+
+        return Requestor.request(RequestMethod.GET, endpoint, params, ClaimCollection.class, client);
```

Comment:
"status" and "type" need to be extracted and stored in the resultant collection.

Reply:
Good catch! Added